tests/CMakeLists.txt

```diff
@@ -1,6 +1,37 @@
 find_package(Catch2 2 REQUIRED)
 find_package(trompeloeil 42 REQUIRED)
 
+# Add a CTest fixture target that runs a CMake build target.
+#
+# Reuses the target name for both the test name and fixture name.
+function(openassetio_add_test_fixture_target target_name)
```

Comment:
[question] By introducing these helpers, do we run the risk of making this harder to understand for newcomers - as it is no longer vanilla CMake? They don't contain a lot of logic. How likely is it that we will need to change what they do over time? (I'm just thinking back to how opaque previous projects have been when they become 50% 'custom functions' vs the build systems native calls).

Reply:
:thinking: It's a reasonable concern. 

I had hoped that ensuring we use an `openassetio_` namespace would add some clarity at least on what functions are added by us. 

There is an example in this PR of having to change what they do (adding a label), which would have been a larger error-prone refactor without these.  Hard to predict how likely it is we'll need to change these functions going forward, though.

Reply:
If it helps to address the concern, this is from the "Functions and Macros" chapter of Professional CMake (written by one of the authors of cmake):

> Looking back on the material covered in this book so far, CMake’s syntax is already starting to look a lot like a programming language in its own right. It supports variables, if-then-else logic, looping and inclusion of other files to be processed. It should be no surprise to learn that CMake also supports the common programming concepts of functions and macros too. Much like their role in other programming languages, functions and macros are the primary mechanism for projects and developers to extend CMake’s functionality and to encapsulate repetitive tasks in a natural way.

Reply:
A reference from USD: https://github.com/PixarAnimationStudios/USD/blob/release/pxr/base/tf/CMakeLists.txt